### PR TITLE
Widen invalid username check, prevent empty usernames

### PR DESCRIPTION
**Pull request by Mystiflow**
Regex allows the following.
- any alphanumerical character
- hyphens and underscores
- limits characters to 1-16, but this is already checked in InitialHandler anyway

### CODE REVIEW

**Comment by Mystiflow** [comment]
Thanks. I didn't know that.

File: proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java [hunk 1]
```diff
@@ -325,9 +325,9 @@ public void handle(LoginRequest loginRequest) throws Exception
         Preconditions.checkState( thisState == State.USERNAME, "Not expecting USERNAME" );
         this.loginRequest = loginRequest;
 
-        if ( getName().contains( "." ) )
+        if ( getName().length() == 0 )
```
Comment (Dimatert9) [inline proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java]: getName().trim().length() ?

File: api/src/main/java/net/md_5/bungee/Util.java [hunk 2]
```diff
@@ -5,13 +5,16 @@
...
 import java.util.UUID;
+import java.util.regex.Pattern;
 
 /**
  * Series of utility classes to perform various operations.
  */
 public class Util
 {
 
+    private static final Pattern VALID_USERNAME_REGEX = Pattern
```
Comment (md-5) [inline api/src/main/java/net/md_5/bungee/Util.java]: No line break

**Comment by md-5** [comment]
Does Spigot do any validation?
Is there any reason we need these checks given that:
1) Doesn't affect online mode (the only real supported config)
2) Offline mode servers should have a plugin already to validate according to their needs (see comments on other languages)

The treatment of . was a hacky workaround several years ago for that causing errors in the config system

**Pull request closed without merging**